components/prism-core.js

```diff
@@ -235,6 +297,8 @@ var _ = self.Prism = {
 		var rest = grammar.rest;
 
 		if (rest) {
+			_.optimizeGrammar(rest);
```

Comment:
That is a good point. I could just concatinate it to the `_competition` property of grammar.
```js
if (rest) {
	_.optimizeGrammar(grammar);
	_.optimizeGrammar(rest);

	for (var token in rest) {
		grammar[token] = rest[token];
	}
	delete grammar.rest;
	grammar._competition.concat(rest._competition);
}
```